### PR TITLE
upgrade go sample to new layout

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description**
- Upgrades the go sample for the new layout
- Add OLM and scorecard manifests as well  

PS.: The steps to test are in the README which was updated in this PR.

### CODE REVIEW

**Comment by camilamacedo86** [comment]
/hold

I upgrade the sample with master and our latest tag release here is v0.18.1. 
we need check if we will or not create a tagged release for 0.19.x and the v1.0.0 here. 

c/c @joelanford

File: go/memcached-operator/bundle/manifests/cache.example.com_memcacheds.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+apiVersion: apiextensions.k8s.io/v1beta1
```
Comment (ravitri) [inline go/memcached-operator/bundle/manifests/cache.example.com_memcacheds.yaml:1]: @camilamacedo86 - As per https://github.com/operator-framework/operator-lifecycle-manager/pull/1416, the CRD defaults to v1 version than v1beta1 and k8s 1.19 will deprecate v1beta1 for CRD as such.. What about defaulting to v1 version of CRD in new layout?

Comment (camilamacedo86) [reply go/memcached-operator/bundle/manifests/cache.example.com_memcacheds.yaml:1]: The problem is that when we run make bundle it is not copying the generated CRD's manifest and is re-creating it with the `v1beta1`. So, it shows a bug. I am raising an issue for that.

Comment (camilamacedo86) [reply go/memcached-operator/bundle/manifests/cache.example.com_memcacheds.yaml:1]: Tks for raise your concern see: https://github.com/operator-framework/operator-sdk/issues/3580

**Comment by camilamacedo86** [comment]
Hi @varshaprasad96,

I will close this one because was made manually and you are working on it the script.

**Pull request closed without merging**